sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go

```diff
@@ -0,0 +1,40 @@
+package localscanner
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/certrepo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	scannerSecretName          = "scanner-tls"            // #nosec G101 not a hardcoded credentials
+	scannerDbSecretName        = "scanner-db-tls"         // #nosec G101 not a hardcoded credentials
+	scannerV4IndexerSecretName = "scanner-v4-indexer-tls" // #nosec G101 not a hardcoded credentials
+	scannerV4DbSecretName      = "scanner-v4-db-tls"      // #nosec G101 not a hardcoded credentials
+)
+
+// NewServiceCertificatesRepo creates a new ServiceCertificatesRepoSecrets that persists certificates for
+// scannerV2, scanner DB, scannerV4 indexer and ScannerV4 DB in k8s secrets.
+// Those secrets have to have ownerReference as the only owner reference.
+func NewServiceCertificatesRepo(ownerReference metav1.OwnerReference, namespace string,
+	secretsClient corev1.SecretInterface) certrepo.ServiceCertificatesRepo {
```

Comment:
I know the code was doing that the same way before the refactor, but returning an interface instead of a concrete type maybe cumbersome in some situations. A general rule is to accept interfaces on the input but always return the concrete types. Do you think, we could apply this rule here?

Reply:
I suppose the reason it returns the interface is because this function is stored in TLSIssuer as a function pointer:
```go
type serviceCertificatesRepoGetter func(ownerReference metav1.OwnerReference, namespace string,
	secretsClient corev1.SecretInterface) certrepo.ServiceCertificatesRepo
```

I could either:
1. leave as it is
2. change the function pointer to also return a concrete type, but then it couples TLSIssuer with the concrete implementation
3. maybe give up the function pointer approach and just store the returned object - I'm not sure of the implications though yet, I suppose there was a reason for the current design

Reply:
Thanks for checking this! 
The decision is up to you. I am fine with any option - including 1.